streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java

```diff
@@ -180,6 +182,19 @@ public void shouldRemoveStreamThread() throws Exception {
         }
     }
 
+    @Test
+    public void shouldnNotRemoveStreamThreadWithTimeout() throws Exception {
+        try (final KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), properties)) {
+            addStreamStateChangeListener(kafkaStreams);
+            startStreamsAndWaitForRunning(kafkaStreams);
+
+            final int oldThreadCount = kafkaStreams.localThreadsMetadata().size();
+            stateTransitionHistory.clear();
+            assertThrows(TimeoutException.class, () -> kafkaStreams.removeStreamThread(Duration.ZERO.minus(DEFAULT_DURATION)));
```

Comment:
It's a bit weird to test this by passing in a negative timeout but I don't have any good ideas for forcing it to exceed the timeout 😕

Reply:
I don't either...

Reply:
But it isn't consistent because if the thread removes itself then the timeout its started

Reply:
Have you thought of using the `KafkaStreams` constructor that takes a time object? You can pass `MockTime` to the constructor and progress time as you want.